File: Go/Summary-Ranges.go
package main

import "fmt"

func summaryRanges(nums []int) []string {
	ranges := []string{}

	start := 0
	for i := 1; i < len(nums); i++ {
		if nums[i-1]+1 != nums[i] {
			ranges = append(ranges, rangeString(nums[start], nums[i-1]))
			start = i
		}
	}

	if len(nums) > 0 {
		ranges = append(ranges, rangeString(nums[start], nums[len(nums)-1]))
	}

	return ranges
}

func rangeString(start, end int) string {
	if start == end {
		return fmt.Sprint(start)
	}
	return fmt.Sprintf("%d->%d", start, end)
}

File: Javascript/Summary-Ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function (nums) {
  let ranges = [];

  let start = 0;
  for (let i = 1; i < nums.length; i++) {
    if (nums[i - 1] + 1 !== nums[i]) {
      ranges.push(rangeString(nums[start], nums[i - 1]));
      start = i;
    }
  }

  if (nums.length > 0) {
    ranges.push(rangeString(nums[start], nums[nums.length - 1]));
  }

  return ranges;
};

function rangeString(start, end) {
  if (start == end) {
    return `${start}`;
  }
  return `${start}->${end}`;
}
